src/sage/rings/lazy_species.py

```diff
@@ -0,0 +1,1395 @@
+r"""
+Lazy Combinatorial Species
+
+We regard a combinatorial species as a sequence of group actions of
+the symmetric groups `\mathfrak S_n`, for `n\in\NN`.
+
+Coefficients of lazy species are computed on demand.  They have
+infinite precision, although equality can only be decided in special
+cases.
+
+AUTHORS:
+
+- Mainak Roy, Martin Rubey, Travis Scrimshaw (2024-2025)
+
+EXAMPLES:
+
+    We can reproduce the molecular expansions from Appendix B in
+    [GL2011]_ with little effort.  The molecular expansion of the
+    species of point determining graphs can be computed as the
+    species of graphs composed with the compositional inverse of the
+    species of non-empty sets.::
```

Comment:
```suggestion
EXAMPLES:

    We can reproduce the molecular expansions from Appendix B in
    [GL2011]_ with little effort.  The molecular expansion of the
    species of point determining graphs can be computed as the
    species of graphs composed with the compositional inverse of the
    species of non-empty sets::
```
Everything before `EXAMPLES:` in this docstring is overindented by 4 spaces.